### ISSUE TITLE
Hybrid (PackageReference and packages.config) projects end up taking 2x the disk space for nugets

### ISSUE THREAD

**Issue by asklar** [microsoft/react-native-windows#9624]
> Also boost takes up a lot of space :). This seems like a mistake that needs to be cleaned up. This test app may not even need the packages.config at all. Start by deleting it and seeing if it still works.

Any machine building MSRN from sources will have to get a copy of boost.
Using `PackageReference`-only projects, there will be only one copy in the developer's machine.

As Nick explained, the "extra" `packages.config` exists for hybrid scenarios because the classic NuGet mechanism almost hard-codes the extracted NuGet's location (usually `$(SolutionDir)packages`).

_Originally posted by @JunielKatarn in https://github.com/microsoft/react-native-windows/issues/9328#issuecomment-1007907442_

**Comment by NickGerleman** [comment]
Why do you think this leads to extra disk space usage?

`packages.config` projects download the package first to the user's NuGet cache, and second, make s copy to the local `packages` directory. `PackageReference` dependencies skip the second step, and the project will instead reference the package in the NuGet cache directly.

**Comment by JunielKatarn** [comment]
Yeah, I was a little confused by the issue's description.

Paraphrasing Nick's reply:
- With `packages.config` only: All dependencies are duplicated (user cache [NuGet's `globalPackagesFolder`]) + clone cache) PER CLONE.
- With hybrid `PackageReference`: Only dependencies of third-party projects that use `packages.config` will be duplicated.

Unless we misunderstood, this is a non-issue and can be closed.

**Comment by asklar** [comment]
@JunielKatarn  IIUC Restore happens twice, once with PackageReference and once w/packages.config style:

```xml
  <!-- Should match entry in $(ReactNativeWindowsDir)vnext\Directory.Build.targets -->
  <!--Allow implicitly restoring NuGet dependencies in C++ projects using the Visual Studio IDE.-->
  <Target Name="BeforeResolveReferences" Condition="'$(BuildingInsideVisualStudio)' == 'true' AND '$(MSBuildProjectExtension)' == '.vcxproj'">
    <!--
      Ensure restoring of PackageReference dependencies.
    -->
    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Restore" Properties="RestoreProjectStyle=PackageReference" />

    <!--
      Ensure restoring of packages.config dependencies.

      RestoreProjectStyle=PackagesConfig    - Required to use the packages.config mechanism
      RestorePackagesConfig=true            - Required to use the packages.config mechanism
      RestoreUseStaticGraphEvaluation=false - Override setting from Microsoft.ReactNative.Common.props
    -->
    <MSBuild Projects="$(MSBuildProjectFile)" Targets="Restore" Properties="RestoreProjectStyle=PackagesConfig;RestorePackagesConfig=true;RestoreUseStaticGraphEvaluation=false" />
  </Target>

```

**Comment by NickGerleman** [comment]
There are two restores. One effects `packages.config` style dependencies (e.g. any current community module), and the other will restore `PackageReference` dependencies.

These are two separate sets of dependencies. In cases where the same dependency is present in both, the same per-user NuGet cache is shared. So there will not be extra disk space usage. For the set of dependencies that was moved from `packages.config`, to PackageReference, we will use less disk space than previously, due to not copying to the project as well.

I agree with @JunielKatarn that this can be closed unless we've observed an issue where there is greater disk usage. Regardless of dual restore, we should be using strictly less disk space than previously.
